src/mlpack/methods/ann/layer/padding_impl.hpp

```diff
@@ -42,10 +46,32 @@ void Padding<InputDataType, OutputDataType>::Forward(
 {
   nRows = input.n_rows;
   nCols = input.n_cols;
-  output = arma::zeros(nRows + padWLeft + padWRight,
-      nCols + padHTop + padHBottom);
-  output.submat(padWLeft, padHTop, padWLeft + nRows - 1,
-      padHTop + nCols - 1) = input;
+  
+  if (inputWidth == 0 || inputHeight == 0)
+  {
+    output = arma::zeros(nRows + padWLeft + padWRight,
+        nCols + padHTop + padHBottom);
+    output.submat(padWLeft, padHTop, padWLeft + nRows - 1,
+        padHTop + nCols - 1) = input;
+  }
+  else
+  {
+    inSize = input.n_elem / (inputWidth * inputHeight * nCols);
+    inputTemp = arma::Cube<eT>(const_cast<arma::Mat<eT>&>(input).memptr(),
+        inputWidth, inputHeight, inSize * nCols, false, false);
+    outputTemp = arma::zeros<arma::Cube<eT> >(inputWidth + padWLeft + padWRight,
```

Comment:
```suggestion
    outputTemp = arma::zeros<arma::Cube<eT>>(inputWidth + padWLeft + padWRight,
```

Reply:
I somewhere read that in such situations you should have a space so your compiler doesn't misunderstands it as a `>>` pipe, I mean this isn't a preoblem with mine but i just wanted to know if that is a thing because i did read it in accelerated c++.

Reply:
I have applied the suggestion though.